src/gluonts/nursery/duru/data.py

```diff
@@ -0,0 +1,488 @@
+        #     def __init__(self):
+        #         self.n_items = 5000
+        #
+        #         self.data = []
+        #
+        #         # noise calculation (first three lines just for the right shape)
+        #         starting_point = np.random.uniform(0, 1) * 2 * np.pi
+        #         x = np.linspace(starting_point + 0, starting_point + 20 * np.pi, 1000)
+        #         sin_x = np.sin(x)
+        #         noise = np.random.normal(loc=0., scale=0.3, size=sin_x.shape)
+        #
+        #         # all items are identical
+        #         for t in range(self.n_items):
+        #             starting_point = np.random.uniform(0, 1) * 2 * np.pi
+        #             x = np.linspace(starting_point + 0, starting_point + 20 * np.pi, 1000)
+        #             sin_x = np.sin(x)
+        #             # add noise
+        #             sin_x = sin_x + noise
+        #             # convert to float32, s.t. later compatible with torch model which expects 32 precision
+        #             sin_x = sin_x.astype(np.float32)
+        #
+        #             item = {}
+        #             item['start'] = 0.  # dummy data
+        #             item['target'] = sin_x
+        #             item['item_id'] = str(t)
+        #             self.data.append(item)
+        #
+        #
+        #     def __iter__(self):
+        #         for t in range(self.n_items):
+        #             # print(t)
+        #             yield self.data[t]
+        #         # raise StopIteration
+        #
+        #     def __len__(self):
+        #         return self.n_items
+        # NEW ----
+
+        dataset_train = SineWaveGluonTSDataset()
+        dataset_val = SineWaveGluonTSDataset()
+        dataset_test = SineWaveGluonTSDataset()
+
+    elif H.dataset == "2gauss":
+
+        class TwoGaussians(gluonts.dataset.Dataset):
+            def __init__(self):
+                self.n_items = 50  # must be divisible by 2
+
+                self.data = []
+
+                # all items are identical; this order of for loops for shuffeling reasons
+                for t in range(int(self.n_items / 2)):
+                    for i in range(2):
+                        if i == 0:
+                            mean = -1.0
+                        else:
+                            mean = 1.0
+                        noise = np.random.normal(
+                            loc=mean, scale=0.3, size=(1000)
+                        )
+                        noise = noise.astype(np.float32)
+
+                        item = {}
+                        item["start"] = 0.0  # dummy data
+                        item["target"] = noise
+                        item["item_id"] = str(t)
+                        self.data.append(item)
+
+            def __iter__(self):
+                for t in range(self.n_items):
+                    # print(t)
+                    yield self.data[t]
+                # raise StopIteration
```

Comment:
Can't this just be
```suggestion
                yield from self.data
```

Reply:
perhaps yes, I haven't tested it, but would leave it for now

Reply:
Then let's just change this.